[jit][fuser] Remove hard-coded NVRTC specific constant from fuser header

bwasti opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22699 [jit][fuser] Remove hard-coded NVRTC specific constant from fuser header**

Differential Revision: [D16192290](https://our.internmc.facebook.com/intern/diff/D16192290)

bwasti committed:
[jit][fuser] Remove hard-coded NVRTC specific constant from fuser header

yinghai commented on torch/csrc/jit/passes/graph_fuser.cpp:
```diff
@@ -138,6 +138,9 @@ struct GraphFuser {
   Symbol kind_ = prim::FusionGroup;
 
+  // Default limit works for nvrtc limitations, change with setInputArgLimit
+  size_t subgraph_arg_limit_ = 128;
```
Maybe we should still copy all the comments.

wanchaol commented on torch/csrc/jit/fuser/compiler.cpp:
```diff
@@ -332,10 +332,6 @@ std::shared_ptr<FusedKernel> compileKernel(
   }
 
-  // Have checked the limit at graph_fuser. Assert nothing else changing that.
-  AT_ASSERT((flat_inputs.size() + flat_outputs.size()) <=
```
Can you elaborate more about the motivation? with this change fusion compiler could not catch the case where it reach over the arguments limit (although it's rare).

zrphercule replied:
we would like to extend the arguments limit here. Not sure if this assertion should be removed.

wanchaol replied:
Then I think we can still leave the `fusion_kernel_args_limit = 128` in the header with no const expr, and set the limit in graph fuser? that way the fusion compiler can still catch the overflow on argument size.

bwasti replied:
the limit is somewhat of a misnomer -- it's actually an NVRTC specific limitation.  Ideally it wouldn't exist at all.

I'm not sure the best approach to ripping it out, but I don't see why the compiler.cpp file would need to have this logic if it's already checked in the fusion pass

wanchaol reviewed (approved):
It looks like checking producer nodes inputs/outputs should be enough already than checking `flat_inputs/outputs` again in the fusion compiler.

yinghai commented:
```
./test/test_torch.py:3447:135: B950 line too long (134 > 120 characters)
```

wanchaol commented:
> ```
> ./test/test_torch.py:3447:135: B950 line too long (134 > 120 characters)
> ```

I think that's a master breakage on linting

bwasti committed:
Update on "[jit][fuser] Remove hard-coded NVRTC specific constant from fuser header"

[jit][fuser] Remove hard-coded NVRTC specific constant from fuser header

gh-metadata: pytorch pytorch 22699 gh/bwasti/27/head
